### PR TITLE
(PUP-8579) Extract yumrepo type and provider

### PR DESCRIPTION
**Pull request by joshcooper**
Yumrepo has been extracted into
https://github.com/puppetlabs/puppetlabs-yumrepo_core

### CODE REVIEW

**Comment by joshcooper** [comment]
Blocked on https://github.com/puppetlabs/puppet-agent/pull/1477

**Comment by joshcooper** [comment]
jenkins please test this

**Comment by melissa** [comment]
```
[root@wn9l66evafmtf9n ~]# yum info puppet-agent
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
Installed Packages
Name        : puppet-agent
Arch        : x86_64
Version     : 5.99.1.43.g8283964
Release     : 1.el7
Size        : 88 M
Repo        : installed
From repo   : pl-puppet-agent-828396465bf709a491cabda9f4120076900fa5a2
Summary     : The Puppet Agent package contains all of the elements needed to run puppet, including ruby, facter, and hiera.
URL         : https://www.puppetlabs.com
License     : See components
Description : The Puppet Agent package contains all of the elements needed to run puppet, including ruby, facter, and hiera.
            :
            : Contains the following components:
            : cpp-hocon 0.1.5
            : cpp-pcp-client 1.5.6-2-gda5cde8
            : dmidecode 2.12
            : facter 3.11.2-25-ge27494e
            : hiera 3.4.3-9-g7dc8f03
            : leatherman 1.4.2-8-gde9e833
            : libwhereami 0.2.1-7-gdb20b4c
            : module-puppetlabs-augeas_core 1.0.0
            : module-puppetlabs-selinux_core 1.0.0
            : module-puppetlabs-yumrepo_core 1.0.0
            : module-puppetlabs-zfs_core 1.0.0
            : module-puppetlabs-zone_core 1.0.0
            : puppet 5.5.2-418-g6badb7d
            : puppet-runtime 201806220
            : pxp-agent 1.9.2-23-ga8b08cc
            : rubygem-puppet-resource_api v1.4.0-6-g6ba4429
            : runtime
            : shellpath 2015-09-18
            : virt-what 1.18
            : wrapper-script
[root@wn9l66evafmtf9n ~]# tree /opt/puppetlabs/puppet/vendor_modules/yumrepo_core/
/opt/puppetlabs/puppet/vendor_modules/yumrepo_core/
├── CHANGELOG.md
├── Gemfile
├── lib
│   └── puppet
│       ├── provider
│       │   └── yumrepo
│       │       └── inifile.rb
│       └── type
│           └── yumrepo.rb
├── metadata.json
├── Rakefile
├── README.md
├── REFERENCE.md
└── spec
    ├── acceptance
    │   ├── nodesets
    │   │   └── default.yml
    │   └── tests
    │       └── yumrepo_spec.rb
    ├── default_facts.yml
    ├── lib
    │   └── puppet_spec
    │       ├── compiler.rb
    │       └── files.rb
    ├── spec_helper_acceptance.rb
    ├── spec_helper_local.rb
    ├── spec_helper.rb
    └── unit
        ├── provider
        │   └── yumrepo
        │       └── inifile_spec.rb
        └── type
            └── yumrepo_spec.rb

15 directories, 18 files
```

File: spec/integration/provider/yumrepo_spec.rb [hunk 1]
```diff
@@ -1,130 +0,0 @@
-#!/usr/bin/env ruby
```
Comment (melissa) [inline spec/integration/provider/yumrepo_spec.rb]: I'm not seeing this file in the yumrepo_core repo. Did it get dropped?

Comment (joshcooper) [reply spec/integration/provider/yumrepo_spec.rb]: Ah yeah, it was overly stubbing things and I moved it to an acceptance test: https://github.com/puppetlabs/puppetlabs-yumrepo_core/commit/9c3b30f2a90b13f7b9676c424a19dc877ac93a53

**Pull request merged**